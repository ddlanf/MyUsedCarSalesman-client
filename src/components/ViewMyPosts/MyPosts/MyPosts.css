.view-my-posts-post{
  width: 40%;
  border-radius: 20px;
  margin: 1%;
  margin-top: 6%;
  background-color: rgba(237, 247, 102, 0.952);
  border: #999090 1px solid;
}

.view-my-posts-car-name{
  text-align: center;
  width: 100%;
  font-size: 3.5vw;
  margin-bottom: 3%;
  font-family: 'Merriweather Sans', sans-serif;
  font-weight: bold;
}

.view-my-posts-car-image-box{
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  height: 35%;
}

.view-my-posts-car-image{
  width: 100%;
  border: #999090 1px solid;
}

.view-my-posts-post > ul{
  list-style: none;
  padding: 0;
  text-align: center;
  margin: 0;
  width: 100%;
}

.view-my-posts-lists  li{
  font-family: 'Merriweather Sans', sans-serif;
}

ul .view-my-posts-car-location{
  font-size: 3vw;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  text-align: left;
  margin-top: -3px;
  margin-bottom: 0;
}

ul .view-my-posts-car-price{
  font-size: 5vw;
  margin: 5%;
}

ul .view-my-posts-car-commission{
  font-size: 4vw;
  padding-left: 5%;
  padding-right: 5%;
  margin: 1%;
  margin-bottom: 10%;
}

.view-my-post_commision-title{
    display: block;
    font-family: 'Merriweather Sans', sans-serif;
    text-decoration: underline;
}

.view-my-posts-lists > li{
    font-size: 3vw;
    margin-bottom: 1%;
}

.view-my-posts-post{
  text-decoration: none;
  color: black;
}

.view-my-posts-button{
  width: 50%;
  max-width: 150px;
  max-height: 50px;
  font-size: 3vw;
  margin-bottom: 5%;
}

.view-my-confirm-delete{
  font-size: 4vw;
  margin-bottom: 8%;
  margin-top: 0;
  color: red;
}

@media screen and (min-width: 1000px) {
  .view-my-posts-post{
    width: 25%;
  }

  .view-my-posts-post:hover{
    background-color: yellow;
    transition: 0.4s;
  }

  .view-my-posts-car-name{
    font-size: 30px;
    margin-bottom: 3%;
  }

  .view-my-posts-lists > li{
    font-size: 30px;
  }

  ul .view-my-posts-car-price{
    font-size: 40px;
    margin: 0;
  }

  ul .view-my-posts-car-commission{
    font-size: 30px;
  }

  ul .view-my-posts-car-location{
    font-size: 20px;
  }

  .view-my-posts-button{
    font-size: 30px;
  }

  .view-my-confirm-delete{
    font-size: 30px;
  }
}