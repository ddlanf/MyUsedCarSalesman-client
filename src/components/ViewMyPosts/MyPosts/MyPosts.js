import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import PostApiService from '../../../services/posts-api-service'
import PostContext from '../../../contexts/PostContext'
import './MyPosts.css'

export default class MyPosts extends Component {

    static contextType = PostContext

    state = { clicked : [] }

    deleteClicked = (id) =>{
        let cloneArray = this.state.clicked
        cloneArray.push(id)
        this.setState({ clicked : cloneArray })
        console.log(cloneArray)
    }

    deletePost = (postId) => {
        this.context.clearError()
        PostApiService.deletePost(postId)
            .then(this.context.deletePost(postId))
            .catch(this.context.setError)
    }

    componentDidMount(){
        this.setState({ clicked: [] })
    }

    rendorPosts(){
        const { images } = this.props
        const thumbNailDefault = [{ "src": "", "alt":"" }]
        const userPosts = this.props.posts

        const posts = userPosts.map(post => {
                let thumbNail = images.filter(image => image.post_id === post.id)[0] || thumbNailDefault;
                return (
                    <div
                        className="view-my-posts-post"
                        to={`/view-my-posts/${post.id}`}
                            key={post.id} 
                        >
                        <p className="view-my-posts-car-name">{post.year} {post.make} {post.model}</p>
                        <div 
                            className="view-my-posts-car-image-box">
                            {<img 
                                className="view-my-posts-car-image"
                                src={thumbNail.src ? thumbNail.src : require('../../../Utils/myusecarsalesman_pics/notfound.png')}
                                alt={thumbNail.alt ? thumbNail.alt : "Not Found"}
                            />}
                        </div>
                        <ul
                            className="view-my-posts-lists">
                            <li>
                                <p className="view-my-posts-car-location">
                                {post.location}
                                </p>
                            </li>
                            <li>
                                <p className="view-my-posts-car-price">
                                    ${post.price}
                                </p>
                            </li>
                            <li>
                                <p className="view-my-posts-car-commission">
                                    <span className="view-my-post_commision-title">Commission</span>
                                    {post.commission_amount}
                                </p>
                            </li>
                            <li>
                                <Link
                                    className="view-my-post-link"
                                    to={`./edit-post/${post.id}`}
                                    >
                                           <button className="view-my-posts-button">Edit</button>
                                </Link>
                            </li>
                            <li>
                                {this.state.clicked.includes(post.id) ?
                                <>
                                <button 
                                    className="view-my-posts-button"
                                    onClick={() => this.deletePost(post.id)}
                                    >
                                    Delete
                                </button>
                                <p className="view-my-confirm-delete">Are you sure?</p>
                                </>
                                :
                                <button 
                                    className="view-my-posts-button"
                                    onClick={() => this.deleteClicked(post.id)}
                                    >
                                    Delete
                                </button>}
                            </li>
                        </ul>
                    </div>
                )
            })

        return posts;
    }

    

    render() {
        return (
            <>
                {this.rendorPosts()}
            </>
        )
    }
}

MyPosts.defaultProps = {
    posts: [],
    images: []
  };


