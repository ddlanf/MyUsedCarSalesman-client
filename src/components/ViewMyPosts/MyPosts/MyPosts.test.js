import React from 'react';
import ReactDOM from 'react-dom';
import MyPosts from './MyPosts';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('MyPosts component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <MyPosts
                          posts={[]}
                          images={[]}
                        />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <MyPosts
                      posts={[]}
                      images={[]}
                    />
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});