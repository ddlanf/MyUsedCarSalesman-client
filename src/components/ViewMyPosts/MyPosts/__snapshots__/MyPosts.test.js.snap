// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPosts component renders the UI as expected 1`] = `null`;
